client/src/controllers/SwapController.ts

```diff
@@ -262,38 +275,113 @@ export class SwapController extends Controller<
     }) as CustomEvent<{ requestUrl: string }>;
 
     if (beginEvent.defaultPrevented) return Promise.resolve();
-
+    const formMethod = this.formElement.getAttribute('method') || undefined;
```

Comment:
Unless it absolutely must be `undefined`?

```suggestion
    const formMethod = this.formElement.getAttribute('method');
```

Reply:
TypeScript's definition is `method?: string`, and `getAttribute()` returns `null` if the attribute doesn't exist. It will complain about being unable to assign `null` to `string | undefined`, and there are a few other ways around this:

- cast the value as `RequestInit['method']` anyway, suppressing TypeScript's complain but actually passing `method: null` in the options for `fetch`
- use `|| 'get'` instead, which is the default but we'll explicitly pass it instead of leaving it undefined
- use `...(formMethod ? { method: formMethod } : {})` or extract the RequestInit into a variable and do `if (formMethod) { init.method = formMethod; }`

Pick your poison? 😂

Reply:
🤡